### ISSUE TITLE
Separate current state from current action in Branching Trees

### ISSUE THREAD

**Issue by PaulRBerg** [sablier-labs/v2-core#621]
## Problem

Copy-pasting @DrakeEvans's feedback from a private chat on Telegram:

> I mostly see the "when" keyword on the branching tree here, do you differentiate between a given state and action in the `.tree` files?  I typically think of tests in the Arrange, Act, Assert phases which correlate to Given, When, Then in Cucumber style notation.  But it looks like you are combining the current state of the system and the current action into a single statement like "when the admin is not the same as the current admin" (line 8).  I think this is a little confusing (or at least it was to me on the first read) because it's not clear in which instances we are talking about function arguments and where we are talking about the state of the system without context clues.  And admittedly this situation wasn't too hard to parse.
>
> Looking at [`transferAdmin`](https://github.com/sablier-labs/v2-core/blob/main/test/unit/concrete/adminable/transfer-admin/transferAdmin.tree):
>
> - On line 2, "the admin" refers to the currently set admin (state of the system)
> - On line 8, "the admin" refers to the function parameter, and "current admin" is referring to the state of the system
>
> To be clear I really like the branching idea for keeping the information DRY, but I wonder if there is a way to make it slightly more explicit without the need for a full description of the prior state which can be verbose.  Some way to delineate between how the action takes place and the current state prior to an action

Drake is absolutely right - our tree terminology commingles the current state (contract state) with the current action (function arguments). It would be helpful to draw a distinction between the two

## Potential Solution

Follow Cucumber's [Gherkin](https://cucumber.io/docs/gherkin/) syntax and replace "when" with "given" for the tree nodes that refer to the contract state.

## Related

- ETHDenver talk: [Better Smart Contracts with Behavior-Driven Development](https://www.youtube.com/watch?v=T70txXsSxig)

**Comment by PaulRBerg** [comment]
Hey @SensationalShubham! This is a good first issue to work on.

Could we do it this way?

1. You start the refactor on the [unit tests](https://github.com/sablier-labs/v2-core/tree/c5f69fb02f258e7924714b263ac25b3e6c1c45fd/test/unit) (and only them)
2. You open a draft PR
3. We review it
4. We continue refactoring the rest of the code base

**Comment by PaulRBerg** [comment]
Telegram - I've the same username as here.

**Comment by PaulRBerg** [comment]
@andreivladbrg - could you pick up this task? You can start with the integration tests for the `Comptroller`.

**Comment by alexfertel** [comment]
@PaulRBerg @andreivladbrg Just to confirm, the change to the syntax of `tree`s would be that some `when`s are now `given`s and that reflects in the contracts as a substitution as well?

**Comment by PaulRBerg** [comment]
Precisely right, @alexfertel.

You can see some related commentary in https://github.com/sablier-labs/v2-core/discussions/647.
